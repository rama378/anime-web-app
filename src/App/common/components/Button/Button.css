.Button {
  cursor: pointer;
  background: transparent;
  border: none;
  padding: 16px 24px;
  border-radius: 32px;
  font-weight: bold;
  font-size: var(--font-size);
  font-family: var(--font-family);
  transition: 0.2s all;
}

.Button svg {
  height: 20px;
  width: 20px;
  transition: 0.2s all;
}

.Button:hover {
  font-size: var(--font-size-large);
  margin: -2px;
  transition: 0.2s all;
}

.Button:hover svg {
  height: 22px;
  width: 22px;
  transition: 0.2s all;
}

.Button:active {
  font-size: var(--font-size);
  opacity: 0.5;
  transition: 0.2s all;
}

.Button:active svg {
  height: 20px;
  width: 20px;
  transition: 0.2s all;
}

.Button--default {
  color: var(--background);
  background-color: var(--text-color);
}

.Button--clear {
  color: var(--text-color);
}
