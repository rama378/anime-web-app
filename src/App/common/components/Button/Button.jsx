import PropTypes from 'prop-types';
import './Button.css';

const Button = (props) => {
	const { children, style, ...rest } = props;

	return (
		<button {...rest} className={`Button Button--${style}`}>{props.children}</button>
	);
};

Button.propTypes = {
	children: PropTypes.any.isRequired,
	style: PropTypes.string,
};

Button.defaultProps = {
	style: 'default',
};

export default Button;
