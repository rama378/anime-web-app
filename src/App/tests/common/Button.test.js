import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Button from '../../common/components/Button/Button';

const renderer = new ShallowRenderer();

describe('<Button />', () => {
  it('should render and match the snapshot no prop given', () => {
    renderer.render(<Button>Button</Button>);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render and match the snapshot given prop style default', () => {
    renderer.render(<Button style="default">Button</Button>);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render and match the snapshot given prop style clear', () => {
    renderer.render(<Button style="clear">Button</Button>);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
