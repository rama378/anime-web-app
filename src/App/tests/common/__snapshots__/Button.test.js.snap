// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render and match the snapshot given prop style clear 1`] = `
<button
  className="Button Button--clear"
>
  Button
</button>
`;

exports[`<Button /> should render and match the snapshot given prop style default 1`] = `
<button
  className="Button Button--default"
>
  Button
</button>
`;

exports[`<Button /> should render and match the snapshot no prop given 1`] = `
<button
  className="Button Button--default"
>
  Button
</button>
`;
